FlightsFile.cs

```diff
@@ -0,0 +1,147 @@
+                if (String.IsNullOrEmpty(newLine))
+                    continue;
+
+                var newFlight = newLine.Split(", ");
+
+                if (!int.TryParse(newFlight[0], out var airplaneCapacity) ||
+                    !decimal.TryParse(newFlight[1], out var price) ||
+                    !DateTime.TryParse(newFlight[3], out DateTime departureDate) ||
+                    !Enum.TryParse(typeof(FlightClass), newFlight[7], out var flightClass) ||
+                    !Flight.IsValid(new Flight(airplaneCapacity, price, newFlight[2],
+                    departureDate, newFlight[4], newFlight[5], newFlight[6],
+                    (FlightClass)flightClass)))
+                {
+                    errorFound = true;
+                    errorLines.Append($"{line}, "); 
+                }
+            }
+
+            if (errorFound)
+            {
+                RemoveAll();
+                throw new Exception(errorLines.ToString() +
+                    "\nMake sure that you follow the constraints.");
+            }
+
+            _fileReader.Close();
+            return true;
+        }
+
+        public static List<Flight> GetAll()
+        {
+            var flightsList = new List<Flight>();
+            var _fileReader = new StreamReader(FlightsFilePath);
+
+            while (!_fileReader.EndOfStream)
+            {
+                var flightData = _fileReader.ReadLine()?.Split(", ");
+
+                // Flights CSV file format: ID, Airplane capacity, Price,
+                // Departure country, Departure date, Departure airport,
+                // Destination country, Arrival airport, Class
+                flightsList.Add(new Flight(int.Parse(flightData[0]), 
+                    int.Parse(flightData[1]), int.Parse(flightData[2]),
+                    flightData[3], DateTime.Parse(flightData[4]), flightData[5],
+                    flightData[6], flightData[7], 
+                    (FlightClass)Enum.Parse(typeof(FlightClass), flightData[8])));
+            }
+
+            _fileReader.Close ();
+            return flightsList;
+        }
+
+        public static Flight Get(int flightID)
+        {
+            if (!Exists(flightID))
+                throw new Exception($"The flight of ID: {flightID} doesn't exist.");
+
+            return GetAll().Single(flight => flight.ID == flightID);
+        }
+
+        public static List<Flight> GetAllAvailable()
+        {
+            if (!HasData)
+                throw new Exception("There are no flights.");
+
+            return GetAll().Where(flight =>
+                {
+                    if (BookingsFile.HasData)
+                    {
+                        var numberOfPassengers = BookingsFile.GetAll()
+                            .Where(booking => booking.Flight.ID == flight.ID)
+                            .Count();
+
+                        return numberOfPassengers < flight.AirplaneCapacity;
+                    }
```

Comment:
u can move this logic to a new method for more readability 
name it IsFlightAvailableForBooking